x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/logic/rule_assets/prebuilt_rule_assets_client/methods/fetch_tags_by_version.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { AggregationsTermsAggregateBase } from '@elastic/elasticsearch/lib/api/types';
+import type { SavedObjectsClientContract, SavedObjectsRawDocSource } from '@kbn/core/server';
+import { invariant } from '../../../../../../../../common/utils/invariant';
+import { PREBUILT_RULE_ASSETS_SO_TYPE } from '../../prebuilt_rule_assets_type';
+import type { RuleVersionSpecifier } from '../../../rule_versions/rule_version_specifier';
+
+/**
+ * Fetches unique tags from prebuilt rule assets for specified rule versions.
+ *
+ * @param savedObjectsClient - Saved Objects client
+ * @param versions - An array of rule version specifiers, each containing a rule_id and version
+ * @returns A promise that resolves to an array of unique tags, sorted alphabetically.
+ */
+export async function fetchTagsByVersion(
+  savedObjectsClient: SavedObjectsClientContract,
+  versions: RuleVersionSpecifier[]
+): Promise<string[]> {
+  const soIds = versions.map(
+    (version) => `${PREBUILT_RULE_ASSETS_SO_TYPE}:${version.rule_id}_${version.version}`
+  );
+
+  const searchResult = await savedObjectsClient.search<
+    SavedObjectsRawDocSource,
+    { unique_tags: AggregationsTermsAggregateBase<{ key: string; doc_count: number }> }
+  >({
+    type: PREBUILT_RULE_ASSETS_SO_TYPE,
+    namespaces: ['default'],
+    _source: false,
+    size: 0,
+    query: {
+      terms: {
+        _id: soIds,
+      },
+    },
+    aggs: {
+      unique_tags: {
+        terms: {
+          field: `${PREBUILT_RULE_ASSETS_SO_TYPE}.tags`,
+          size: 10_000,
```

Comment:
nit: Could we avoid magic constants? Ideally we should use a well known constant. This constant should have a good description why 10K is the number.